src/test/groovy/pl/sda/rentacar/domain/rent/RentServiceIntegrationSpec.groovy

```diff
@@ -0,0 +1,297 @@
+import org.springframework.test.context.ContextConfiguration
+import pl.sda.rentacar.domain.car.BodyType
+import pl.sda.rentacar.domain.car.Car
+import pl.sda.rentacar.domain.car.CarCreateRequest
+import pl.sda.rentacar.domain.car.CarService
+import pl.sda.rentacar.domain.car.CarStatus
+import pl.sda.rentacar.domain.client.Client
+import pl.sda.rentacar.domain.client.ClientCreateRequest
+import pl.sda.rentacar.domain.client.ClientService
+import pl.sda.rentacar.domain.department.Department
+import pl.sda.rentacar.domain.department.DepartmentCreateRequest
+import pl.sda.rentacar.domain.department.DepartmentService
+import pl.sda.rentacar.domain.employee.Employee
+import pl.sda.rentacar.domain.employee.EmployeeCreateRequest
+import pl.sda.rentacar.domain.employee.EmployeeService
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.time.LocalDate
+
+@SpringBootTest
+@ContextConfiguration
+class RentServiceIntegrationSpec extends Specification {
+
+    @Autowired
+    RentService service
+
+    @Autowired
+    RentRepository repository
+
+    @Autowired
+    ClientService clientService
+
+    @Autowired
+    DepartmentService departmentService
+
+    @Autowired
+    EmployeeService employeeService
+
+    @Autowired
+    CarService carService
+
+    @Shared
+    def clientDaniel = new Client(
+            1L,
+            "Daniel",
+            "Olbrychski",
+            "kochamskutery@szabelka.eu",
+            "777888999"
+    )
+
+    @Shared
+    def departmentZadupie = new Department(
+            1L,
+            "Zadupie",
+            [] as Set,
+            [] as Set)
+
+    @Shared
+    def employeeMax = new Employee(
+            1L,
+            "Max",
+            "Wersztapen",
+            departmentZadupie)
+
+    @Shared
+    def carForFamily = new Car(
+            1L,
+            "Renaul",
+            "Espace",
+            2007,
+            CarStatus.AVAILABLE,
+            BigDecimal.valueOf(110),
+            BodyType.MINIVAN
+    )
+    @Shared
+    def departmentRequest = new DepartmentCreateRequest(
+            "Radom",
+            [] as Set,
+            [] as Set
+    )
+
+    @Shared
+    def clientRequest = new ClientCreateRequest(
+            "Otylia",
+            "Jedrzejczak",
+            "motylek@o2.pl",
+            "555777666")
+
+    @Shared
+    def clientRequest2 = new ClientCreateRequest(
+            "Michal",
+            "Zewlakow",
+            "wybitnyreprezentant@pzpn.pl",
+            "555777888")
+
+    @Shared
+    def carRequest = new CarCreateRequest(
+            "Mazda",
+            "MX-5",
+            2019,
+            BigDecimal.valueOf(100L),
+            BodyType.ROADSTER
+    )
+
+    @Shared
+    def carRequest2 = new CarCreateRequest(
+            "Volvo",
+            "C70",
+            2012,
+            BigDecimal.valueOf(100L),
+            BodyType.CONVERTIBLE
+    )
+
+    def 'should add rent'() {
+        given:
+        cleanup()
```

Comment:
Calling `cleanup` manually is redundant. Please checkout out https://spockframework.org/spock/docs/1.0/spock_primer.html
cleanup will be called automatically after each test method